InvenTree/common/notifications.py

```diff
@@ -108,13 +110,26 @@ class EmailNotification(BulkNotificationMethod):
     ]
 
     def get_targets(self):
+        """
+        Return a list of target email addresses,
+        only for users which allow email notifications
+        """
+
+        allowed_users = []
+
+        for user in self.targets:
```

Comment:
Could we just use a filter? Seems like this could start taking a long time with a lot of targets.

Reply:
We have to individually check if each user has email notifications active.. Can we do that with a filter?

Reply:
Not in reach of my computer atm. But it should be possible with a reference filter. So you filter the queryset on the target users, annotate the queryset with the reference to the setting value and then filter on the annotated value. That should result in only 1 database hit, not 1 per target user.

Reply:
I'm happy to look at optimizing this if you can propose something better :)